### PR TITLE
fix(ZNTA-2107) word breaking and removal of hyphens for editor

### PR DESCRIPTION
**Pull request by kathryngo**
JIRA issue URL: https://zanata.atlassian.net/browse/ZNTA-2107

Fixed GWT bug where words broke over multiple lines and were hyphenated. Because this is a high priority fix for translators I have made the changes to the release branch.

word-wrapping

## Reviewer tasks

The following is based on the
[Zanata Development Guidelines](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines).
As a reviewer, you should check the guidelines regularly for updates, and just
to refresh your memory.

The reviewer can use this list for reference to make sure
they have considered all the important points.

Use the checkboxes or not, whatever works best for you.

 - Code quality
   - [ ] Code passes style check (checkstyle/eslint)
   - [ ] Code is clear and concise
   - [ ] UI code is internationalised
   - [ ] Code has adequate comments where appropriate
   - Code is in an appropriate place
     - [ ] Classes are in appropriate packages
     - [ ] Code fits with class's responsibilities (SRP)
   - [ ] Configuration files are documented enough
   - If code is removed
     - [ ] No remaining code references the removed code
       - [ ] No orphaned rewrite rules
       - [ ] No orphaned config settings
     - [ ] No remaining comments refer to the removed code
     - [ ] Unused imports and libraries are removed
 - Testing
   - [ ] New code has tests
   - [ ] Changed code has tests
   - [ ] All tests pass
   - [ ] Test coverage stable or increased
 - Documentation
   - [ ] New features are documented
   - [ ] Docs removed for deleted features
   - [ ] Docs updated for all changed features
   - [ ] Developer/sysadmin docs updated if appropriate
 - If there are new dependencies
   - [ ] Does each new dependency pass all the
         [Considerations for new dependencies](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines#new-technologies-and-dependencies)?

----
*This template can be updated in .github/PULL_REQUEST_TEMPLATE.md*

---
This change is [Reviewable](https://reviewable.io/reviews/zanata/zanata-platform/420)

### CODE REVIEW

**Comment by aeng** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/420#-:-KooDX2NWYhZ4BaqGrIh:bnfp4nl)*

File: server/zanata-frontend/src/frontend/app/styles/style.less [hunk 1]
```diff
@@ -6035,6 +6035,14 @@ i.i--left.i--lock, i.i--left.i--trash {
   padding-bottom:@spacing-base-and-a-half;
 }
 
+.translationContainer pre.cm-s-default {
+  word-break: break-word;
```
Comment (djansen-redhat) [inline server/zanata-frontend/src/frontend/app/styles/style.less]: I think this should be break-all or normal (break-word appears to be a word-wrap value?)

Comment (kathryngo) [inline server/zanata-frontend/src/frontend/app/styles/style.less]: break-all does not work. break-word is a webpack specific value. Does break-all work for you?

Comment (djansen-redhat) [inline server/zanata-frontend/src/frontend/app/styles/style.less]: Your code seems to work in Chrome, but not Firefox.

**Comment by kathryngo** [comment]
I can fix this easily if I make changes to the assets repo because I can remove any weird overrides and conflicts. Shou I go ahead and do that? I am unsure what the current status of assets is and if I should make changes there.

**Comment by kathryngo** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[server/zanata-frontend/src/frontend/app/styles/style.less, line 6039 at r1](https://reviewable.io:443/reviews/zanata/zanata-platform/420#-Koo07VsdKu_5Nz4wjxS-r1-6039:-Kooq3pXyz8nLOmx5c6M:b-2orw2k) ([raw file](https://github.com/zanata/zanata-platform/blob/be8512c3d229c5eed71f4697db19b61c9f2ba123/server/zanata-frontend/src/frontend/app/styles/style.less#L6039)):*
<details><summary><i>Previously, djansen-redhat (Damian Jansen) wrote…</i></summary><blockquote>

Your code seems to work in Chrome, but not Firefox. 
</blockquote></details>

Aha. Can you add another line Rsaying the following after the word break saying
'word-wrap: break-all' 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/420)*

**Comment by kathryngo** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[server/zanata-frontend/src/frontend/app/styles/style.less, line 6039 at r1](https://reviewable.io:443/reviews/zanata/zanata-platform/420#-Koo07VsdKu_5Nz4wjxS-r1-6039:-KooqpWbEfn6ir2wXFq1:bxrrdsi) ([raw file](https://github.com/zanata/zanata-platform/blob/be8512c3d229c5eed71f4697db19b61c9f2ba123/server/zanata-frontend/src/frontend/app/styles/style.less#L6039)):*
<details><summary><i>Previously, kathryngo wrote…</i></summary><blockquote>

Aha. Can you add another line Rsaying the following after the word break saying
'word-wrap: break-all' 


</blockquote></details>

Also add 
White-space:pre-wrap

I'm afk atm but I will look at it in a few hours

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/420)*

**Comment by kathryngo** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[server/zanata-frontend/src/frontend/app/styles/style.less, line 6039 at r1](https://reviewable.io:443/reviews/zanata/zanata-platform/420#-Koo07VsdKu_5Nz4wjxS-r1-6039:-KooqpWbEfn6ir2wXFq1:bxrrdsi) ([raw file](https://github.com/zanata/zanata-platform/blob/be8512c3d229c5eed71f4697db19b61c9f2ba123/server/zanata-frontend/src/frontend/app/styles/style.less#L6039)):*
<details><summary><i>Previously, kathryngo wrote…</i></summary><blockquote>

Aha. Can you add another line Rsaying the following after the word break saying
'word-wrap: break-all' 


</blockquote></details>

Also add 
White-space:pre-wrap

I'm afk atm but I will look at it in a few hours

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/420)*

**Comment by djansen-redhat** [comment]
*[server/zanata-frontend/src/frontend/app/styles/style.less, line 6039 at r1](https://reviewable.io:443/reviews/zanata/zanata-platform/420#-Koo07VsdKu_5Nz4wjxS-r1-6039:-KooupTU3kdFyKm6Z-4U:b-2ugaeh) ([raw file](https://github.com/zanata/zanata-platform/blob/be8512c3d229c5eed71f4697db19b61c9f2ba123/server/zanata-frontend/src/frontend/app/styles/style.less#L6039)):*
<details><summary><i>Previously, kathryngo wrote…</i></summary><blockquote>

Also add 
White-space:pre-wrap

I'm afk atm but I will look at it in a few hours
</blockquote></details>

word-break:break-all and white-space:pre-wrap had no effect. When I added word-wrap:normal, the lines wrapped correctly. http://imgur.com/a/v6Fk7

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/420)*

**Comment by kathryngo** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[server/zanata-frontend/src/frontend/app/styles/style.less, line 6039 at r1](https://reviewable.io:443/reviews/zanata/zanata-platform/420#-Koo07VsdKu_5Nz4wjxS-r1-6039:-KooyT49bo9gG8FE9P64:b5fodua) ([raw file](https://github.com/zanata/zanata-platform/blob/be8512c3d229c5eed71f4697db19b61c9f2ba123/server/zanata-frontend/src/frontend/app/styles/style.less#L6039)):*
<details><summary><i>Previously, djansen-redhat (Damian Jansen) wrote…</i></summary><blockquote>

word-break:break-all and white-space:pre-wrap had no effect. When I added word-wrap:normal, the lines wrapped correctly. http://imgur.com/a/v6Fk7
</blockquote></details>

Ok thanks.  I won't be able to make changes until tomorrow but it's top priority. Deathly sick right now 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/420)*

**Comment by kathryngo** [comment]
@djansen-redhat it now works in Firefox and Chrome. I just needed to play with the word-break settings a bit more. Ready for you to test again.

**Comment by djansen-redhat** [comment]
✅ Looks good to me. Hope you feel better soon.

**Comment by kathryngo** [comment]
@djansen-redhat need you to approve the review so I can get all green

**Comment by aeng** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/420)*

**Comment by kathryngo** [comment]
Still need @djansen-redhat to review...

**Comment by djansen-redhat** [comment]
@kathryngo sorry for the delay

**Comment by kathryngo** [comment]
@djansen-redhat no worries at all. 👍

**Pull request merged**